clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -295,7 +296,7 @@ public void onFailure(RuntimeException e) {
      * be polled to see if the request completed successfully.
      * @return A request future whose completion indicates the result of the JoinGroup request.
      */
-    private RequestFuture<ByteBuffer> sendJoinGroupRequest() {
+    private RequestFuture<ByteBuffer> performGroupJoin() {
```

Comment:
The function does both JoinGroup and SyncGroup, so the name seemed a little misleading. Maybe I can be explicit and call it `performJoinAndSync`? Or maybe `joinNextGeneration`?